lib/properties.js

```diff
@@ -296,7 +293,7 @@ internals.properties.prototype.parseString = function(property, joiObj) {
       'x-format'
     );
 
-    internals.convertRules(property, describe.rules, ['lowercase', 'uppercase', 'trim'], 'x-convert');
+    internals.convertRules(property, describe.rules, ['case', 'trim'], 'x-convert');
```

Comment:
Most of these changes are matches to the updated `joi` spec. Everything is a clear extension now. lowercase and uppercase both call the case function with a key of `lower` or `upper`. The tests reflect this.